### PR TITLE
[Merged by Bors] - chore: remove unused `set_option in`

### PR DESCRIPTION
**Pull request by adomani**
Found by the linter #13653.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by adomani** [comment]
!bench

File: Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean [hunk 1]
```diff
@@ -295,7 +295,6 @@ alias hasSum_log_sub_log_of_abs_lt_1 := hasSum_log_sub_log_of_abs_lt_one
 #adaptation_note /-- after v4.7.0-rc1, there is a performance problem in `field_simp`.
 (Part of the code was ignoring the `maxDischargeDepth` setting:
  now that we have to increase it, other paths becomes slow.) -/
```
Comment (adomani) [inline Mathlib/Analysis/SpecialFunctions/Log/Deriv.lean:297]:
```suggestion
```

File: Mathlib/Analysis/SpecialFunctions/Stirling.lean [hunk 2]
```diff
@@ -113,7 +113,6 @@ theorem log_stirlingSeq_diff_le_geo_sum (n : ℕ) :
 #adaptation_note /-- after v4.7.0-rc1, there is a performance problem in `field_simp`.
 (Part of the code was ignoring the `maxDischargeDepth` setting:
  now that we have to increase it, other paths become slow.) -/
```
Comment (adomani) [inline Mathlib/Analysis/SpecialFunctions/Stirling.lean:115]:
```suggestion
```

File: Mathlib/MeasureTheory/Covering/BesicovitchVectorSpace.lean [hunk 3]
```diff
@@ -337,7 +337,6 @@ variable [NormedSpace ℝ E]
 #adaptation_note /-- after v4.7.0-rc1, there is a performance problem in `field_simp`.
 (Part of the code was ignoring the `maxDischargeDepth` setting:
  now that we have to increase it, other paths becomes slow.) -/
```
Comment (adomani) [inline Mathlib/MeasureTheory/Covering/BesicovitchVectorSpace.lean:339]:
```suggestion
```

File: Mathlib/NumberTheory/ModularForms/SlashActions.lean [hunk 4]
```diff
@@ -96,7 +96,6 @@ local notation:100 f " ∣[" k "]" γ:100 => ModularForm.slash k γ f
 #adaptation_note /-- after v4.7.0-rc1, there is a performance problem in `field_simp`.
 (Part of the code was ignoring the `maxDischargeDepth` setting:
  now that we have to increase it, other paths become slow.) -/
```
Comment (adomani) [inline Mathlib/NumberTheory/ModularForms/SlashActions.lean:98]:
```suggestion
```

**Review by adomani — COMMENTED** [review]
Drag along `#adaptation_note`s with the removal.

**Comment by grunweg** [comment]
Thanks for doing this! The code itself looks good. I looked at the benchmark results in detail (checking the files with a `set_option removed`): most are fine or get slightly faster, with one or two exceptions:
- `RootsOfUnity` gets very slightly slower (170 Mio); overall instructions are neutral/better (+700 Mio)
- `Haar.OfBasis` gets appreciably slower (44,2 -> 45,6*10⁹ instructions, for a single line change): can you profile or benchmark that file locally? If the slow-down persists, I would rather leave the set_option and leave a comment that this is kept for performance purposes.

Feel free to maintainer merge on my behalf once you have done so.

**Comment by adomani** [comment]
Here are the local heartbeat counts:
```lean
Roots of unity
-- no `set_option tactic.skipAssignedInstances false in`
Used 16863 heartbeats, which is less than the current maximum of 200000.

-- with `set_option tactic.skipAssignedInstances false in`
Used 16154 heartbeats, which is less than the current maximum of 200000.
```
---
```lean
Haar of Basis
-- no `set_option tactic.skipAssignedInstances false in`
Used 48121 heartbeats, which is less than the current maximum of 200000.
-- with `set_option tactic.skipAssignedInstances false in`
Used 42718 heartbeats, which is less than the current maximum of 200000.
```
Using `set_option trace.profiler true` with and without the option shows that `Basis.prod_parallelepiped` benefits from the option, since elaboration goes from ~5 to ~4 seconds.  The following declarations in the file take much less than a second with or without the option.

Thus, removing `set_option tactic.skipAssignedInstances false in` results in a 1 second slowdown in the declaration.

I really do not know what is the cut-off time limit for preferring to add a `set_option`.

**Review by Ruben-VandeVelde — APPROVED** [review]
lgtm

maintainer merge

**Comment by mattrobball** [comment]
bors merge

**Pull request closed without merging**